src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```diff
@@ -694,4 +715,87 @@ private void cancelElectionsForReferenceIds(List<String> referenceIds) {
+    DataAccessRequest dar = collection.getMostRecentDar();
+    List<User> distinctUsers = getDistinctAdminAndChairUsersForDAR(dar);
+    User researcher = userDAO.findUserById(collection.getCreateUserId());
+    if (researcher == null) {
+      logWarn(
+          "Sending new DAR Collection message: Could not find researcher for specified user id: "
+              + collection.getCreateUserId());
+    }
+    String researcherName = researcher == null ? "Unknown" : researcher.getDisplayName();
+    // Only do this for the DAR... dacDAO.findDacsForDatasetIds(dar.getDatasetIds())
+    Collection<Dac> dacsInDAR = dacDAO.findDacsForDatasetIds(dar.getDatasetIds());
+    // Use only the datasets from the dar
+    List<Integer> datasetIds = dar.getDatasetIds();
+    List<Dataset> datasetsInDAR =
+        datasetIds.isEmpty() ? List.of() : datasetDAO.findDatasetsByIdList(datasetIds);
+
+    Map<String, List<String>> sendList = new HashMap<>();
+    for (User user : distinctUsers) {
+      List<Dac> matchingDacsForUser = getMatchingDacs(user, dacsInDAR);
+      for (Dac dac : matchingDacsForUser) {
+        List<String> matchingDatasetsForDac = getMatchingDatasets(dac, datasetsInDAR);
+        if (matchingDatasetsForDac != null) {
+          sendList.put(dac.getName(), matchingDatasetsForDac);
+        }
+      }
+      // If the dar is not a progress report, use the DAR template else use the PR template.
+      if (dar.getProgressReport()) {
+        // Use the reference ID to link the fact that this progress report will have been noted.
+        // the DAR Code at this point will be ambiguous.
+        emailService.sendNewProgressReportRequestEmail(user, sendList, researcherName, collection.getDarCode(), dar.getReferenceId());
+      } else {
+        emailService.sendNewDARRequestEmail(user, sendList, researcherName, collection.getDarCode());
+      }
+    }
+  }
+
+  private List<User> getDistinctAdminAndChairUsersForDAR(DataAccessRequest dar) {
+    List<Integer> datasetIds = dar.getDatasetIds();
+    return getDistinctAdminAndChairUsersForDatasetIds(datasetIds);
+  }
+
+  private List<User> getDistinctAdminAndChairUsersForDatasetIds(List<Integer> datasetIds) {
+    List<User> admins = userDAO.describeUsersByRoleAndEmailPreference(UserRoles.ADMIN.getRoleName(),
+        true);
+    Set<User> chairPersons = userDAO.findUsersForDatasetsByRole(datasetIds,
+        Collections.singletonList(UserRoles.CHAIRPERSON.getRoleName()));
+    // Ensure that admins/chairs are not double emailed
+    // and filter users that don't want to receive email
+    return Streams.concat(admins.stream(), chairPersons.stream())
+        .filter(u -> Boolean.TRUE.equals(u.getEmailPreference()))
+        .distinct()
+        .toList();
+  }
+
+  private List<Dac> getMatchingDacs(User user, Collection<Dac> dacsInDAR) {
+    List<Integer> dacIDs = user.getRoles().stream()
+        .filter(ur -> ur.getDacId() != null)
+        .map(UserRole::getDacId)
+        .toList();
```

Comment:
Slightly less code to call `getDacId` first
```suggestion
        .map(UserRole::getDacId)
        .filter(Objects::nonNull)
        .toList();
```